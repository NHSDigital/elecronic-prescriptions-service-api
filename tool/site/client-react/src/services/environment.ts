export type Environment = "internal-dev" | "int"

export const internalDev = "internal-dev"
export const int = "int"

export function isDev(environment: Environment): boolean {
  return environment === internalDev
}

export function isInt(environment: Environment): boolean {
  return environment === int
}
