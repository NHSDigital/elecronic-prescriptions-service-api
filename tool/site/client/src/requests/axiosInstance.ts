import axios from "axios"

export const axiosInstance = axios.create({
  validateStatus: () => true
})

// todo: refresh token if required
axiosInstance.interceptors.request.use(function (config) {
  return config
}, function (error) {
  return Promise.reject(error)
})
