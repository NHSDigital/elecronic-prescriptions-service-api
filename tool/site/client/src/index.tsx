import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
