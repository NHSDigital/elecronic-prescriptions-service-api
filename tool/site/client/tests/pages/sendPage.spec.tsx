import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import SendPage from "../../src/pages/sendPage"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {internalDev} from "../../src/services/environment"

const baseUrl = "baseUrl/"
const token = "MzQxMWJmMjUtMDNlMy00N2FiLWEyOGItMGIyYjVlNTg4ZGU3"
const prescriptionId = "003D4D-A99968-4C5AAJ"
const context: AppContextValue = {baseUrl, environment: internalDev}

const downloadSignaturesUrl = `${baseUrl}sign/download-signatures`
const sendUrl = `${baseUrl}api/prescribe/send`

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test.skip("Displays confirmation page if single prescription is sent successfully", async () => {
  moxios.stubRequest(downloadSignaturesUrl, {
    status: 200,
    response: {
      prescription_id: prescriptionId,
      bundle_id: "1",
      success: "unknown"
    }
  })

  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      prescription_id: prescriptionId,
      bundle_id: "1",
      request: "JSON Request",
      request_xml: "XML Request",
      response: "JSON Response",
      response_xml: "XML Response",
      success: true
    }
  })

  const container = await renderPage()

  await waitFor(() => screen.getByText(prescriptionId))

  expect(screen.getByText(JSON.stringify("JSON Request"))).toBeTruthy()
  expect(screen.getByText("XML Request")).toBeTruthy()
  expect(screen.getByText(JSON.stringify("JSON Response"))).toBeTruthy()
  expect(screen.getByText("XML Response")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays confirmation page if multiple prescriptions are sent successfully", async () => {
  moxios.stubRequest(downloadSignaturesUrl, {
    status: 200,
    response: {
      results: [
        {
          prescription_id: "003D4D-A99968-4C5AAJ",
          success: true
        },
        {
          prescription_id: "008070-A99968-41CD9V",
          success: true
        },
        {
          prescription_id: "010E34-A99968-467D9Z",
          success: true
        }
      ]
    }
  })

  const container = await renderPage()

  expect(screen.getByText("003D4D-A99968-4C5AAJ")).toBeTruthy()
  expect(screen.getByText("008070-A99968-41CD9V")).toBeTruthy()
  expect(screen.getByText("010E34-A99968-467D9Z")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<SendPage token={token}/>, context)
  await waitFor(() => screen.getByText(/Send Result/))
  return container
}
