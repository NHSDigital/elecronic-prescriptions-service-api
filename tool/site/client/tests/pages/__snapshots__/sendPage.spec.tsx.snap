// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays confirmation page if multiple prescriptions are sent successfully 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Send Results</label></h1>
<div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy Prescription IDs</button></div>
<div class=\\"nhsuk-table-responsive\\">
  <table class=\\"nhsuk-table\\">
    <thead class=\\"nhsuk-table__head\\">
      <tr class=\\"nhsuk-table__row\\">
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Bundle ID</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Prescription ID</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Success</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\"></th>
      </tr>
    </thead>
    <tbody class=\\"nhsuk-table__body\\">
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">1</td>
        <td class=\\"nhsuk-table__cell\\">003D4D-A99968-4C5AAJ</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
        <td class=\\"nhsuk-table__cell\\">
          <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/view?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
              </svg><span class=\\"nhsuk-action-link__text\\">View</span></a></div>
        </td>
      </tr>
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">2</td>
        <td class=\\"nhsuk-table__cell\\">008070-A99968-41CD9V</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
        <td class=\\"nhsuk-table__cell\\">
          <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/view?prescription_id=008070-A99968-41CD9V\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
              </svg><span class=\\"nhsuk-action-link__text\\">View</span></a></div>
        </td>
      </tr>
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">3</td>
        <td class=\\"nhsuk-table__cell\\">010E34-A99968-467D9Z</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
        <td class=\\"nhsuk-table__cell\\">
          <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/view?prescription_id=010E34-A99968-467D9Z\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
              </svg><span class=\\"nhsuk-action-link__text\\">View</span></a></div>
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Displays confirmation page if single prescription is sent successfully 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Send Result <svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
    </svg></label></h1>
<dl class=\\"nhsuk-summary-list\\">
  <div class=\\"nhsuk-summary-list__row\\">
    <dt class=\\"nhsuk-summary-list__key\\">ID</dt>
    <dd class=\\"nhsuk-summary-list__value\\">003D4D-A99968-4C5AAJ</dd>
  </div>
</dl>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/prescribe/cancel?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Cancel prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/release?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Release prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/view?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">View prescription</span></a></div>
<div class=\\"nhsuk-expander-group\\">
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">1 - FHIR Request that a supplier would send to the FHIR API</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%7B%7D\\" download=\\"message\\">Download</a></div>
      <div class=\\"react-json-view\\" style=\\"font-family: monospace; cursor: default; background-color: rgb(255, 255, 255); position: relative; margin-top: 10px;\\">
        <div class=\\"pretty-json-container object-container\\">
          <div class=\\"object-content\\">
            <div class=\\"object-key-val\\"><span><span style=\\"display: inline-block; cursor: pointer;\\"><div class=\\"icon-container\\" style=\\"display: inline-block; width: 17px;\\"><span class=\\"collapsed-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); height: 1em; width: 1em; padding-left: 2px;\\" viewBox=\\"0 0 15 15\\" fill=\\"currentColor\\"><path d=\\"M0 14l6-6-6-6z\\"></path></svg></span></div><span></span><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184);\\">{</span></span></span><span class=\\"brace-row\\"><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184); padding-left: 0px;\\">}</span>
            <div style=\\"display: inline-block; padding: 0px 0px 0px 10px;\\" class=\\"object-meta-data\\"><span class=\\"copy-to-clipboard-container\\" title=\\"Copy to clipboard\\" style=\\"vertical-align: top; display: none;\\"><span style=\\"cursor: pointer; display: inline;\\"><span class=\\"copy-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;\\" viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\"><g><path d=\\"m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z\\"></path></g></svg></span></span></span></div></span>
          </div>
        </div>
      </div>
    </div>
</div>
</details>
<details class=\\"nhsuk-details nhsuk-expander\\">
  <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">2 - Equivalent hl7v3 that the FHIR API has sent to spine</span></summary>
  <div class=\\"nhsuk-details__text\\">
    <div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Request\\" download=\\"message\\">Download</a></div><pre class=\\"sc-eCImPb WCiLo\\">XML Request</pre>
  </div>
</details>
<details class=\\"nhsuk-details nhsuk-expander\\">
  <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">3 -  The hl7v3 spine response that the FHIR API received from spine</span></summary>
  <div class=\\"nhsuk-details__text\\">
    <div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Response\\" download=\\"message\\">Download</a></div><pre class=\\"sc-eCImPb WCiLo\\">XML Response</pre>
  </div>
</details>
<details class=\\"nhsuk-details nhsuk-expander\\">
  <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">4 - FHIR Response that a supplier will receive from the FHIR API</span></summary>
  <div class=\\"nhsuk-details__text\\">
    <div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%7B%7D\\" download=\\"message\\">Download</a></div>
    <div class=\\"react-json-view\\" style=\\"font-family: monospace; cursor: default; background-color: rgb(255, 255, 255); position: relative; margin-top: 10px;\\">
      <div class=\\"pretty-json-container object-container\\">
        <div class=\\"object-content\\">
          <div class=\\"object-key-val\\"><span><span style=\\"display: inline-block; cursor: pointer;\\"><div class=\\"icon-container\\" style=\\"display: inline-block; width: 17px;\\"><span class=\\"collapsed-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); height: 1em; width: 1em; padding-left: 2px;\\" viewBox=\\"0 0 15 15\\" fill=\\"currentColor\\"><path d=\\"M0 14l6-6-6-6z\\"></path></svg></span></div><span></span><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184);\\">{</span></span></span><span class=\\"brace-row\\"><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184); padding-left: 0px;\\">}</span>
          <div style=\\"display: inline-block; padding: 0px 0px 0px 10px;\\" class=\\"object-meta-data\\"><span class=\\"copy-to-clipboard-container\\" title=\\"Copy to clipboard\\" style=\\"vertical-align: top; display: none;\\"><span style=\\"cursor: pointer; display: inline;\\"><span class=\\"copy-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;\\" viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\"><g><path d=\\"m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z\\"></path></g></svg></span></span></span></div></span>
        </div>
      </div>
    </div>
  </div>
  </div>
</details>
</div>
<div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>"
`;

exports[`Result summary fails if a response is a fail 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Send Results</label></h1>
<div class=\\"sc-dkPtRN jXpeDI\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy Prescription IDs</button></div>
<div class=\\"nhsuk-table-responsive\\">
  <table class=\\"nhsuk-table\\">
    <thead class=\\"nhsuk-table__head\\">
      <tr class=\\"nhsuk-table__row\\">
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Bundle ID</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Prescription ID</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Success</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\"></th>
      </tr>
    </thead>
    <tbody class=\\"nhsuk-table__body\\">
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">1</td>
        <td class=\\"nhsuk-table__cell\\">003D4D-A99968-4C5AAJ</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
        <td class=\\"nhsuk-table__cell\\">
          <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/view?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
              </svg><span class=\\"nhsuk-action-link__text\\">View</span></a></div>
        </td>
      </tr>
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">2</td>
        <td class=\\"nhsuk-table__cell\\">008070-A99968-41CD9V</td>
        <td class=\\"nhsuk-table__cell\\">
          <div style=\\"margin: 0px; float: left;\\" class=\\"nhsuk-loader\\"><span style=\\"width: 34px; height: 34px;\\" class=\\"nhsuk-loader__spinner nhsuk-loader--small\\"></span></div>
        </td>
        <td class=\\"nhsuk-table__cell\\">
          <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/view?prescription_id=008070-A99968-41CD9V\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
              </svg><span class=\\"nhsuk-action-link__text\\">View</span></a></div>
        </td>
      </tr>
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">3</td>
        <td class=\\"nhsuk-table__cell\\">010E34-A99968-467D9Z</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__cross\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path d=\\"M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z\\"></path>
            <path d=\\"M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z\\"></path>
          </svg></td>
        <td class=\\"nhsuk-table__cell\\"></td>
      </tr>
    </tbody>
  </table>
</div>"
`;
